test/metabase/query_processor_test/remapping_test.clj

```diff
@@ -320,3 +320,37 @@
                             [2 123 "Mediocre Wooden Bench"  "Mediocre Wooden Bench"]
                             [3 105 "Fantastic Wool Shirt"   "Fantastic Wool Shirt"]]
                            (mt/rows (qp/process-query q3))))))))))))))
+
+(defmacro with-card-with-query
+  "Like [[mt/with-temp]] for card but run the query to get the result_metadata and create a card with it.
+  Bind the created card to `card-binding`."
+  {:style/indent 2}
+  [query [card-binding & [card-option]] & body]
+  `(mt/with-temp ~Card [card# (merge {:dataset_query   ~query
+                                      :result_metadata (get-in (qp/process-query ~query)
```

Comment:
Could you elaborate? I fail to see why we shouldn't do it. 
Does it add any overhead or something?

Reply:
In general you have to have very good reasons for evaluating a parameter expression twice. Why? Because people generally expect macro calls to behave like function calls and might pass something side effecting as parameter thinking it will be evaluated once. For example. if the expression `query` prints something, your macro will print it twice. This probably not what the users of your macro expect.

Reply:
ahhh
That makes lots of sense. I'll fix it.
Thanks!